mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1009,8 +1064,17 @@ async fn gen_taker_funding_spend_preimage<T: UtxoCommonOps>(
 
     let fee = match fee {
         FundingSpendFeeSetting::GetFromCoin => {
-            coin.get_htlc_spend_fee(DEFAULT_SWAP_TX_SPEND_SIZE, &FeeApproxStage::WithoutApprox)
-                .await?
+            let calculated_fee = coin
+                .get_htlc_spend_fee(DEFAULT_SWAP_TX_SPEND_SIZE, &FeeApproxStage::WithoutApprox)
+                .await?;
+            let taker_instructed_fee =
+                sat_from_big_decimal(&args.taker_payment_fee, coin.as_ref().decimals).mm_err(|e| e.into())?;
+            // If calculated fee is less than instructed fee, use instructed fee because it was never intended
+            // to be deposited to us (maker). If the calculated fee is larger, we will incur the fee difference
+            // just to make sure the transaction is confirmed quickly.
+            // FIXME: Possible abuse vector is that the taker can always send the fee to be just above 90% of the
```

Comment:
> Shouldn't they have already agreed on the fee at the negotiation stage?

Yup, but this is assuming that the fees stayed constant or dropped. This is in place so that we favor succeeding swaps more than refunds.